src/codegen/utilities/interpreter_helpers.py

```diff
@@ -54,67 +57,69 @@ def get_interpreter_functions(metadata):
     """Gets function call arguments."""
-    # This implementation assumes that an array parameter is immediately followed
-    # by the array size when making the c function call.
     function_call_args = []
     size_values = {}
-    SizeParameter = collections.namedtuple("SizeParameter", ["name", "size"])
-    for param in function_metadata.interpreter_parameters:
+    interpreter_parameters = get_interpreter_parameters(function_metadata)
+    for param in interpreter_parameters:
         if param.has_explicit_buffer_size:
             if param.direction == "in":
-                size_values[param.size.value] = SizeParameter(
-                    param.parameter_name, f"len({param.parameter_name})"
-                )
+                size_values[param.size.value] = f"len({param.parameter_name})"
             elif param.direction == "out":
                 if param.size.mechanism == "ivi-dance":
-                    size_values[param.size.value] = SizeParameter(param.parameter_name, "temp_size")
-                if (
+                    size_values[param.size.value] = "temp_size"
+                elif (
                     is_custom_read_write_function(function_metadata)
-                    and param.size.mechanism == "passed-in"
+                    and param.has_explicit_buffer_size
                 ):
-                    size_values[param.size.value] = SizeParameter(
-                        param.parameter_name, f"{param.parameter_name}.size"
-                    )
-
-    for param in function_metadata.interpreter_parameters:
-        if param.direction == "in":
+                    if param.size.mechanism == "passed-in":
+                        size_values[param.size.value] = f"{param.parameter_name}.size"
+
+    for param in interpreter_parameters:
+        if param.parameter_name in size_values:
+            function_call_args.append(size_values[param.parameter_name])
+        elif param.parameter_name == "reserved":
+            function_call_args.append("None")
+        elif param.direction == "in":
             function_call_args.append(param.parameter_name)
         elif param.direction == "out":
             if param.has_explicit_buffer_size:
                 function_call_args.append(param.parameter_name)
             else:
                 function_call_args.append(f"ctypes.byref({param.parameter_name})")
-        if param.has_explicit_buffer_size:
-            if param.size.value in size_values:
-                size_parameter = size_values[param.size.value]
-                if param.parameter_name == size_parameter.name:
-                    function_call_args.append(size_parameter.size)
+    return function_call_args
 
-    # C function call for reserved parameter for read/write function is passed as None.
-    if function_metadata.python_codegen_method == "CustomCode_Read_Write":
-        function_call_args.append("None")
 
-    return function_call_args
+def get_arguments_type(functions_metadata):
```

Comment:
Naming: `get_argument_types`

It returns multiple types. The word "argument" modifies "type(s)".